### PR TITLE
TST/BUG: fix incorrectly-passing Exception in test_html

### PR DESCRIPTION
**Pull request by jbrockmendel**
A `raise` is indented one indent further than it should be.  Fixing this surfaces a failing test that is fixed by decoding bytes before passing it to bs4.

### CODE REVIEW

File: pandas/io/html.py [hunk 1]
```diff
@@ -591,9 +591,14 @@ def _setup_build_doc(self):
     def _build_doc(self):
         from bs4 import BeautifulSoup
 
-        return BeautifulSoup(
-            self._setup_build_doc(), features="html5lib", from_encoding=self.encoding
-        )
+        bdoc = self._setup_build_doc()
+        if isinstance(bdoc, bytes) and self.encoding is not None:
```
Comment (jreback) [inline pandas/io/html.py]: this is very strange

so this sometimes returns bytes? and str otherwise?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/html.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/io/html.py]: i dont know this code particularly well, found this while tracking down `except Exception`.  AFAICT sometimes bs4 wants bytes

**Comment by jbrockmendel** [comment]
gentle ping.  not a blocker, but it is a bugfix so it'd be nice to get it in for the RC

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. can you rebase just in case. ping on green.

**Comment by jbrockmendel** [comment]
rebased+green

**Comment by jreback** [comment]
thanks

**Pull request merged**